derive_generic_visitor/build.rs

```diff
@@ -0,0 +1,8 @@
+use std::{env, fs, path::PathBuf};
+
+fn main() {
+    let readme = fs::read_to_string("../README.md").unwrap();
+    let doc = readme.split_once("<!-- start-doc -->").unwrap().1.trim();
+    let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
+    fs::write(out_dir.join("docs.md"), doc).unwrap();
+}
```

Comment:
Not a huge fan but I see why you did it like this. Could you add the right directive to ensure that this re-runs whenever the README changes? As-is the output may stay cached.

Reply:
Yeah, that's a bit overwhelming.
I don't mind if you want to revert it.

Otherwise, that's a good point, I added that clause!